tests/integration/test_charm.py

```diff
@@ -34,3 +41,107 @@ async def test_is_active(ops_test: OpsTest):
         timeout=300,
     )
+
+
+def _safe_load_file_to_text(filename: str):
+    """Returns the contents of filename if it is an existing file, else it returns filename"""
+    try:
+        text = Path(filename).read_text()
+    except FileNotFoundError:
+        text = filename
+    return text
+
+
+def create_all_from_yaml(
+    yaml_file: str,
+    lightkube_client: lightkube.Client = None,
+):
+    """Creates all k8s resources listed in a YAML file via lightkube
+    Args:
+        yaml_file (str or Path): Either a string filename or a string of valid YAML.  Will attempt
+                                 to open a filename at this path, failing back to interpreting the
+                                 string directly as YAML.
+        lightkube_client: Instantiated lightkube client or None
+    """
+
+    yaml_text = _safe_load_file_to_text(yaml_file)
+
+    if lightkube_client is None:
+        lightkube_client = lightkube.Client()
+
+    for obj in codecs.load_all_yaml(yaml_text):
+        try:
+            lightkube_client.create(obj)
+        except lightkube.core.exceptions.ApiError as e:
+            raise ValueError(f"unable to create resource from file {yaml_file} becuase of {e}")
+
+
+def delete_all_from_yaml(
+    yaml_file: str, lightkube_client: lightkube.Client = None, ignore_errors: bool = True
+):
+    """Deletes all k8s resources listed in a YAML file via lightkube
+    Args:
+        yaml_file (str or Path): Either a string filename or a string of valid YAML.  Will attempt
+                                 to open a filename at this path, failing back to interpreting the
+                                 string directly as YAML.
+        lightkube_client: Instantiated lightkube client or None
+    """
+    log.info("Clearing the namespace")
+    yaml_text = _safe_load_file_to_text(yaml_file)
+
+    if lightkube_client is None:
+        lightkube_client = lightkube.Client()
+
+    for obj in codecs.load_all_yaml(yaml_text):
+        try:
+            lightkube_client.delete(type(obj), obj.metadata.name)
+        except lightkube.core.exceptions.ApiError as e:
+            if ignore_errors:
+                logger.info(f"Ignoring exception {e}")
+                continue
+            raise ValueError(f"unable to delete resource from file {yaml_file}")
+
+
+@pytest.fixture(scope="session")
+def lightkube_client() -> lightkube.Client:
+    """Initiates the lightkube client with PodDefault crd resource"""
+    client = lightkube.Client()
+    create_namespaced_resource(
+        group="kubeflow.org", version="v1alpha1", kind="PodDefault", plural="poddefaults"
+    )
+    return client
+
+
+def test_namespace_selector_poddefault_service_account_token_mounted(lightkube_client):
+    try:
+        sleep(30)  # to overcome this bug https://bugs.launchpad.net/juju/+bug/1981833
+        workloads_file = "./tests/integration/poddefault_test_workloads.yaml"
+        create_all_from_yaml(workloads_file, lightkube_client)
```

Comment:
This yaml creation might be a little safer in a fixture instead of here, but I'm not sure.  The only thing I'm wondering is whether if someone `ctrl+c`'s in the middle of this if the `finally` will happen.  I don't think it does (thus orphaning the resources on the cluster)?  But I **think** (am not sure) that doing `ctrl+c` here if you're using a fixture will still let pytest clean up the fixture.  Do you know?

Reply:
added fixture